lib/parsers/cr3_parser.rb

```diff
@@ -0,0 +1,49 @@
+class FormatParser::CR3Parser
+  include FormatParser::IOUtils
+  require_relative 'cr3_parser/decoder'
```

Comment:
Do we _need_ to require inside the class definition?

Reply:
Yep, otherwise we get the following in FormatParser::CR3Parser::Decoder
'''
Failure/Error:
  class FormatParser::CR3Parser::Decoder < FormatParser::ISOBaseMediaFileFormat::Decoder
    include FormatParser::EXIFParser

    protected

    ATOM_PARSERS = ATOM_PARSERS.merge({
      'CMT1' => :cmt1
    })
    CANON_METADATA_CONTAINER_UUID = '85c0b687820f11e08111f4ce462b6a48'


NameError:
  uninitialized constant FormatParser::CR3Parser

  class FormatParser::CR3Parser::Decoder < FormatParser::ISOBaseMediaFileFormat::Decoder
                    ^^^^^^^^^^^
  Did you mean?  FormatParser::CR2Parser
                 FormatParser::ARWParser
'''

Reply:
I meant, wouldn't it make more sense to put it at the top of the file?

Reply:
Yes, that's what we get when it's at the top of the file. I think it's analagous to a circular import in Python if it's at the top of the file - if it's at the top, it tries to require the decoder, but the decoder then tries to declare itself under `FormatParser::CR3Parser` which hasn't yet been declared because it's waiting to require the decoder which hasn't yet been declared because it's waiting for the parser etc.

Reply:
Got it. Normally, you'd require things from a central place, so you wouldn't run into too much trouble doing it all over the place. But that's a topic for another discussion ✌️